### ISSUE TITLE
"UNALLOCATED_STATS_POINTS" notification for players under level 10

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#9952]
When a player who has not yet unlocked the class system levels up, they get a notification that they have a Stat Point to allocate, even though they cannot yet allocate Points. 

There's [a few problems](https://github.com/HabitRPG/habitica/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aopen%20sort%3Aupdated-desc%20in%3Atitle%20points) with Stat Point allocation at the moment but I think this one is high priority because it will be affecting all new players and will be confusing for them while they're trying to get familiar with the site.

Can the notifications be suppressed when the user isn't meant to get them? I.e., that's when `flags.classSelected` is false (user is too low-level to have chosen a class) or `preferences.disableClasses` is true (user has opted out of the class system).

I'm marking this as help wanted but it's possible that the staff will fix it themselves soon so anyone interested in working on it should comment here and wait for a response before starting (that's always important but more so in this case). 

Below is a bug report and the data showing this problem.

Merry (8d818045-eaa7-421e-a6a4-86a7429d622f): "Hello. I have just levelled up and was told I had a new point to allocate. When I go to my stats, however, there appears to be no way to allocate it. The bit at the bottom where you have the option to add points is simply missing. I cannot allocate points from either desktop or app at the moment. I have tried syncing and also reloading. Many thanks"

```

{
    "_id": "8d818045-eaa7-421e-a6a4-86a7429d622f",
    "stats": {
        "class": "warrior",
        "lvl": 3
    },
    "preferences": {
        "disableClasses": false
    },
    "flags": {
        "classSelected": false
    },
    "notifications": [
        {
            "type": "UNALLOCATED_STATS_POINTS",
            "seen": false,
            "data": {
                "points": 1
            },
            "id": "84babb88-0b5c-4ca9-97eb-9daed8b2aa48"
        }
    ],
    "auth": {
        "timestamps": {
            "created": "2018-01-28T13:47:56.849Z"
        }
    }
}
 ```

**Comment by citrusella** [comment]
For what it's worth, I already don't get unallocated stat point messages as an opted-out user.

**Comment by paglias** [comment]
Hmm, I added this check to prevent showing the notification to users who shouldn't have it

```
        if (notification.type === 'UNALLOCATED_STATS_POINTS') {
          if (!this.user.flags.classSelected || this.user.preferences.disableClasses) return false;
        }
```

Which I thought was enough, but I'll check again later. Note that the notification is still added to the user but shouldn't be displayed

**Comment by paglias** [comment]
I'm having some problem replicating this, I just tried to level up on a level 6 account and didn't get any notification. Is the report maybe from mobile? But anyway mobile shouldn't be using the new notifications

cc @vIiRuS

**Comment by paglias** [comment]
So mobile doesn't show notifications (although Android still shows the stat allocation menu when you don't have a class) but I'm unable to replicate this on web. So maybe this is an old report or from mobile? @Alys 

In the meantime I'm adding the `needs investigation` label

**Comment by Alys** [comment]
All I have is what's in the top post. I didn't do any tests myself because there were too many other bug guild reports that day. Amplitude might tell you what platform the user was on at the time.

I was assuming that the stats point notification appearing in the user's notification object was the bug. If we can stop it appearing in the notification object when the user isn't meant to see it, then no client code is needed in the website, both apps, and any third-party tools that display notifications.

**Comment by lemoness** [comment]
Have we had any other reports of this issue?

**Comment by Alys** [comment]
It's definitely reproduceable that a user who has never selected a class is given a notification in their user document in the database, meaning that all three official clients and any relevant third-party tools need to suppress showing it. If it's possible to prevent creating the raw notification data when `preferences.disableClasses` is true or `flags.classSelected` is false, then I'd like to see that done as a low priority fix at least.

**Comment by paglias** [comment]
I'm going to prevent the notification being added to users with no class selected
